src/services/cherry-picker.ts

```diff
@@ -301,24 +301,47 @@ export class CherryPicker {
     // The app/node with the highest success rate and the lowest average latency will
     // be 10 times more likely to be selected than a node that has had failures.
     let weightFactor = 10
+    let previousNodeLatency = 0
+    let latencyDifference = 0
+
+    // This multiplier is tested to produce a curve that adequately punishes slow nodes
+    const weightMultiplier = 15
 
     for (const sortedLog of sortedLogs) {
+      // Set the benchmark from the fastest node
+      if (!previousNodeLatency) {
+        previousNodeLatency = sortedLog.averageSuccessLatency
+      } else {
+        latencyDifference = sortedLog.averageSuccessLatency - previousNodeLatency
+      }
+
+      // The amount you subtract here from the weight factor should be variable based on how
+      // far off this node's average elapsedTime is from the fastest node.
+      // Previously this value was hardcoded 2 in the first bucket
+      if (latencyDifference) {
+        weightFactor = weightFactor - Math.round(latencyDifference * weightMultiplier)
```

Comment:
Just to understand the quick mafs... how big on average would this `latencyDifference` be? also, I guess that as the logs are "sorted" we don't need to do `Math.abs()` on this right?

I haven't proven it mathematically but I feel we might wanna normalize these numbers somehow? Or am I missing something in the maths?

Reply:
Correct they are sorted so no fear of <0. 

Normalize them how?  Need more explanation.

Reply:
> Correct they are sorted so no fear of <0.

sweet! 
> Normalize them how? Need more explanation.

yup scrap that, it mainly applied if things weren't sorted.

on another angle re: huga disparity in latencies:

i'm mainly thinking of edge cases e.g some node being off by 1-2s in latency. How would the formula respond in this case? I'm wondering if there should be some "threshold" were nodes get punished harshly because it might be better for the user to go to fallbacks rather than use the painfully slow nodes. This of course messes with the tokenomics but I'd like to know what you think.

Reply:
I *think* my comments in the code are pretty good? 

If the latencies are

```
node 1  0.25ms
node 2  0.31ms
node 3  0.42ms
node 4  0.45ms
node 5  1.5ms
```

With node 5 being awful, the distribution would be:

```
node1 10
node2 9
node3 8
node4 8
node5 1
```

Reply:
This gives the node a chance to get back in the running without them being selected often.

Reply:
Haha yeah, I see. That's fair:).